File: src/java/src/algorithm/searching/BinarySearch.java
package algorithm.searching;

import algorithm.Solution;

import java.util.Arrays;
import java.util.Scanner;

/**
 * 이분탐색
 * input:
 * 8 32
 * 23 87 65 12 57 32 99 81
 *
 * output:
 * 3
 */
public class BinarySearch implements Solution {

    @Override
    public void solve() {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int target = sc.nextInt();

        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        Arrays.sort(arr);

        int lt = 0;
        int rt = n-1;

        int answer = 0;
        while (lt <= rt) {
            int mid = (rt+lt)/2;
            if (arr[mid] == target) {
                answer = mid+1;
                break;
            } else if (arr[mid] < target) {
                lt = mid+1;
            } else {
                rt = mid-1;
            }
        }

        System.out.println(answer);
    }
}

File: src/kotlin/src/main/kotlin/search/BinarySearch.kt
package search

import java.util.*

fun search(arr: IntArray, target: Int): Int {
    var start = 0
    var end = arr.size - 1

    while(start <= end) {
        var mid = (start + end) / 2
        when {
            arr[mid] == target -> return 1
            arr[mid] < target -> start = mid + 1
            else -> end = mid - 1
        }
    }
    return 0
}

fun main() {
    val sc = Scanner(System.`in`)

    val n =  sc.nextInt()
    val arr = IntArray(n) { sc.nextInt() }
    arr.sort()

    val m = sc.nextInt()
    val targets = IntArray(m) { sc.nextInt() }

    targets.forEach {
        println(search(arr, it))
    }
}